module/metrics/execution.go

```diff
@@ -1081,3 +1082,44 @@ func (ec *ExecutionCollector) ExecutionComputationResultUploaded() {
 func (ec *ExecutionCollector) ExecutionComputationResultUploadRetried() {
 	ec.computationResultUploadRetriedCount.Inc()
 }
+
+type ExecutionCollectorWithTransactionCallback struct {
+	*ExecutionCollector
+	TransactionCallback func(
+		dur time.Duration,
+		stats module.TransactionExecutionResultStats,
+		info module.TransactionExecutionResultInfo,
+	)
+}
+
+func (ec *ExecutionCollector) WithTransactionCallback(
+	callback func(
+		time.Duration,
+		module.TransactionExecutionResultStats,
+		module.TransactionExecutionResultInfo,
+	),
+) *ExecutionCollectorWithTransactionCallback {
+	// if callback is nil, use a no-op callback
+	if callback == nil {
+		callback = func(
```

Comment:
should this support a nil callback? the only difference between `ExecutionCollector` and `ExecutionCollectorWithTransactionCallback` is the callback. it seems strange to allow creating one without a callback

Reply:
The original idea was to only optionally collect transaction execution metrics, but I forgot to add that. I added it now.